### PR TITLE
Robustness fixes: version parsing, payload handling, and input validation

### PR DESCRIPTION
**Pull request by frbuceta**
A batch of small robustness fixes, each independent but too small to justify separate PRs:

- **Firmware version parsing**: versions are now parsed as integer tuples instead of `float`. `float("3.10") == 3.1` only works while Venus OS keeps two-digit minors; tuples are unambiguous and handle `v4.x`, patch releases like `v3.50.1`, and `~beta` suffixes.
- **`unwrap_enum` / `unwrap_bitmask`**: the `data["value"]` access now happens inside the `try` block, so a payload without a `value` key returns `None` like every other unwrapper instead of leaking `KeyError` into the MQTT message handler.
- **`WritableMetric.set()` validation**: numeric values are validated against the resolved min/max range and dropdown values against the available labels, raising a clear `ValueError` instead of silently publishing invalid values to the device. (`step` is deliberately not validated — float arithmetic makes it fragile for little value.)
- **MQTT auth with username only**: auth is now configured when a username *or* password is provided; previously a username without password was silently ignored.
- **`Hub.publish()`**: raises `NotConnectedError` / `ValueError` instead of `assert`, which disappears under `python -O`. Asserts on internal invariants elsewhere were left as-is on purpose.
- **User callback errors**: failures to schedule `on_update` callbacks are logged with `_LOGGER.exception` instead of debug level, so user callback issues are visible in production.

Tests extended to cover the new validation and payload edge cases.

### CODE REVIEW

**Comment by tomer-w** [comment]
How did you find those issues? Is it AI generated?

**Comment by frbuceta** [comment]
> How did you find those issues? Is it AI generated?

Yes, we also use it at work to carry out safety checks. What’s more, I’m a user of the Home Assistant add-on, so if I can help you improve it… we’ll all benefit.

File: victron_mqtt/hub.py [hunk 1]
```diff
@@ -397,8 +398,10 @@ def publish(self, topic_short_id: str, device_id: str, value: str | float | int
         if topic_desc is None:
             _LOGGER.error("No active topic found for topic_short_id: %s", topic_short_id)
             raise TopicNotFoundError(f"No active topic found for topic_short_id: {topic_short_id}")
-        assert self._installation_id is not None, "Installation ID must be set before publishing"
-        assert device_id is not None, "Device ID must be provided"
+        if self._installation_id is None:
```
Comment (tomer-w) [inline victron_mqtt/hub.py]: If those are input parameter checks I would bring them to the top of the function just for readability sake.

File: victron_mqtt/hub.py [hunk 2]
```diff
@@ -725,16 +728,16 @@ def _device_depth(dev: Device) -> tuple[int, int]:
                         ver_str = version_metric.value[1:]
                         if "~" in ver_str:
```
Comment (tomer-w) [inline victron_mqtt/hub.py:729]: I am already handling the ~beta thing so what is exactly broken with the version checks?

File: victron_mqtt/metric.py [hunk 3]
```diff
@@ -318,7 +318,7 @@ def _handle_message(
                 # If the event loop is running, schedule the callback
                 self._hub._loop.call_soon_threadsafe(self._on_update, self, self.value)
             except Exception as exc:
-                log_debug("Error calling callback %s", exc, exc_info=True)
+                _LOGGER.exception("Error scheduling on_update callback for %s: %s", self.unique_id, exc)
```
Comment (tomer-w) [inline victron_mqtt/metric.py:324]: I remember that in some rare timing issues this can be called if things are just going down. This is why it was logged as debug only. Can you verify your fix is safe?

File: victron_mqtt/hub.py (hunk 1 above)
Comment (frbuceta) [reply victron_mqtt/hub.py]: Done in c2f6457 — both checks are now at the top of `publish()`, before the topic lookup.

File: victron_mqtt/hub.py (hunk 2 above)
Comment (frbuceta) [reply victron_mqtt/hub.py:729]: You're right, the `~` split already covers the beta suffixes — that part was never the issue. What `float()` can't handle is:

1. **Any version with a third component** (a `v3.50.1`-style patch release): `float("3.50.1")` raises `ValueError`, so the version falls into the error path and stays at 0.0 — a perfectly good firmware would trigger the "reduced functionality" warning.
2. **The comparison is only correct by convention.** `float("3.50")` is `3.5`, so ordering works today only because Venus minors always have exactly two digits — nothing enforces that. Integer tuples compare exactly regardless of the format (`v4.0`, `v3.50.1`, ...).

So to be fair: nothing misbehaves with the versions Venus publishes *today*. The change removes the format assumptions so this check doesn't need revisiting if the format ever changes.

File: victron_mqtt/metric.py (hunk 3 above)
Comment (frbuceta) [reply victron_mqtt/metric.py:324]: Good catch — I verified it and the timing issue you remember is real: `is_running()` is checked on the MQTT thread, so the loop can close between that check and `call_soon_threadsafe`, which then raises `RuntimeError` during shutdown. In c2f6457 that specific case is logged at debug again (as before), and `_LOGGER.exception` is kept only for unexpected errors.
